### PR TITLE
feat: BlockStreamManager impl for Block Streams

### LINKED ISSUE #15104: Separate the functionality around `BlockStreamManager` for `develop`

### PR DESCRIPTION
**Pull request by derektriley**
**Description**:

Splitting the changes for classes in relation to BlockStreamManager from https://github.com/hashgraph/hedera-services/pull/14579.

**Related issue(s)**:

Fixes #15104

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by mhess-swl — DISMISSED** [review]

**Review by petreze — DISMISSED** [review]
Looks good

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleWorkflow.java [hunk 1]
```diff
@@ -173,6 +182,11 @@ public HandleWorkflow(
         this.kvStateChangeListener = requireNonNull(kvStateChangeListener);
         this.boundaryStateChangeListener = requireNonNull(boundaryStateChangeListener);
         this.migrationStateChanges = requireNonNull(migrationStateChanges);
+        // Temporary flag to control stream mode during transition to block streams
+        STREAM_MODE = configProvider
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleWorkflow.java:186]: We should not use this static variable

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleWorkflow.java [hunk 2]
```diff
@@ -94,6 +98,8 @@ public class HandleWorkflow {
     private static final Logger logger = LogManager.getLogger(HandleWorkflow.class);
 
     public static final String ALERT_MESSAGE = "Possibly CATASTROPHIC failure";
+    // Temporary flag to control stream mode during transition to block streams
+    public static StreamMode STREAM_MODE = RECORDS;
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleWorkflow.java:102]: Need to delete this and use the config like how it is in develop

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleWorkflow.java (hunk 1 above)
Comment (derektriley) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleWorkflow.java]: Thanks, updated.

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleWorkflow.java (hunk 2 above)
Comment (derektriley) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleWorkflow.java]: Thanks, updated.

**Review by mhess-swl — APPROVED** [review]

**Review by petreze — APPROVED** [review]

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM! Thanks @derektriley

**Pull request merged**